packages/api-server/src/methods/modules/eth.ts

```diff
@@ -724,46 +731,74 @@ export class Eth {
       return apiTx;
     }
 
-    // if null, find pending transactions
-    const godwokenTxWithStatus = await this.rpc.getTransaction(gwTxHash);
-    if (godwokenTxWithStatus == null) {
-      return null;
-    }
-    const godwokenTxReceipt = await this.rpc.getTransactionReceipt(gwTxHash);
-    const tipBlock = await this.query.getTipBlock();
-    if (tipBlock == null) {
-      throw new Error("tip block not found!");
-    }
-    let ethTxInfo = undefined;
-    try {
-      ethTxInfo = await filterWeb3Transaction(
-        ethTxHash,
-        this.rpc,
-        tipBlock.number,
-        tipBlock.hash,
-        godwokenTxWithStatus.transaction,
-        godwokenTxReceipt
-      );
-    } catch (err) {
-      logger.error("filterWeb3Transaction:", err);
-      logger.info("godwoken tx:", godwokenTxWithStatus);
-      logger.info("godwoken receipt:", godwokenTxReceipt);
-      throw err;
+    // 2. If null, find pending transactions
+    const ethTxHashKey = ethTxHashCacheKey(ethTxHash);
+    const gwTxHash: Hash | null = await this.cacheStore.get(ethTxHashKey);
+    if (gwTxHash != null) {
+      const godwokenTxWithStatus = await this.rpc.getTransaction(gwTxHash);
+      if (godwokenTxWithStatus == null) {
+        return null;
+      }
+      const godwokenTxReceipt = await this.rpc.getTransactionReceipt(gwTxHash);
+      const tipBlock = await this.query.getTipBlock();
+      if (tipBlock == null) {
+        throw new Error("tip block not found!");
+      }
+      let ethTxInfo = undefined;
+      try {
+        ethTxInfo = await filterWeb3Transaction(
+          ethTxHash,
+          this.rpc,
+          tipBlock.number,
+          tipBlock.hash,
+          godwokenTxWithStatus.transaction,
+          godwokenTxReceipt
+        );
+      } catch (err) {
+        logger.error("filterWeb3Transaction:", err);
+        logger.info("godwoken tx:", godwokenTxWithStatus);
+        logger.info("godwoken receipt:", godwokenTxReceipt);
+        throw err;
+      }
+      if (ethTxInfo != null) {
+        const ethTx = ethTxInfo[0];
+        return ethTx;
+      }
     }
-    if (ethTxInfo != null) {
-      const ethTx = ethTxInfo[0];
-      return ethTx;
+
+    // 3. Find by auto create account tx
+    // TODO: delete cache store if dropped by godwoken
+    // convert to tx hash mapping store if account id generated ?
+    const polyjuiceRawTx = await this.cacheStore.get(cacheKey);
+    if (polyjuiceRawTx != null) {
+      const tipBlock = await this.query.getTipBlock();
+      if (tipBlock == null) {
+        throw new Error("tip block not found!");
+      }
+      // Convert polyjuice tx to api transaction
+      const { tx, fromAddress }: AutoCreateAccountCacheValue =
+        JSON.parse(polyjuiceRawTx);
+      const apiTransaction: EthTransaction =
+        polyjuiceRawTransactionToApiTransaction(
+          tx,
+          ethTxHash,
+          tipBlock.hash,
+          tipBlock.number,
+          fromAddress
+        );
```

Comment:
Can we reuse [`filterWeb3Transaction`](https://github.com/nervosnetwork/godwoken-web3/blob/4d3dddc5881fedf0dbb8cce8a250aa423f9f5e99/packages/api-server/src/filter-web3-tx.ts#L31). Can you explain why if not?

Reply:
We don't have `L2Transaction` now ....